bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedReadChannel.java

```diff
@@ -64,10 +71,22 @@ public int read(ByteBuf dest, long pos) throws IOException {
         return read(dest, pos, dest.writableBytes());
     }
 
+    @Override
+    public long size() throws IOException {
+        if (sealed) {
+            if (fileSize == -1) {
```

Comment:
nit: if the file is already "sealed", should we read the size in the constructor? This way we wouldn't have any concurrency issues here if 2 threads call `size()` at the size time.

Reply:
The `size()` would throw IOException, I don't want to make the constructor announce the exception, so make the fileSize double check to avoid concurrency.